packages/google-auth/README.md

```diff
@@ -0,0 +1,177 @@
+
+## Usage
+In your Google Cloud App settings add the **Redirect uri**. This middleware handles the redirect uri internally as the route you are using the middleware on, so if you decide to use the middleware on the route `/api/v1/auth/google/` the redirect uri will be `DOMAIN/api/v1/auth/google`.
+```
+app.use(
+  "api/v1/auth/google", // -> redirect_uri by default
+  googleAuth({ ... })
+)
+```
+
+Also, there is two ways to use this middleware:
+```
+app.use(
+	'/google',
+	googleAuth({ ... })
+)
+
+app.get(
+	'/google',
+	(c) => {
+		// Handle authentication
+	}
+)
+
+export default app
+```
+
+Or 
+```
+app.get(
+	'/google',
+	googleAuth({ ... }),
+	(c) => {
+		// Handle authentication
+	}
+)
+
+export default app
+
+```
+
+### googleAuth
+```
+import { Hono } from  'hono'
+import { GoogleAuthVariables,googleAuth } from  '@hono/open-auth'
+
+const  app  =  new  Hono<{ Variables:  GoogleAuthVariables }>()
+
+app.use(
+  '/google',
+  googleAuth({
+    client_id:  Bun.env.GOOGLE_ID,
+		client_secret:  Bun.env.GOOGLE_SECRET,
+		response_type: "code",
+		scope: ["openid", "email", "profile"],
+		include_granted_scopes:  true,
+		state:  "sdygz76x-sd3gds2",
+  })
+)
+
+export  default  app
+```
+#### Parameters
+- `client_id`:
+ 	- Type: `string`.
+	- `Required`.
+	-  Your app client ID. You can find this value in the API Console [Credentials page](https://console.developers.google.com/apis/credentials). <br />When developing **Cloudflare Workers**, there's no need to send this parameter. Just declare it in the `wrangler.toml` file as `GOOGLE_ID=`.
+- `client_secret`:
+	- Type: `string`.
+	- `Required`.
+	- Your app client secret. You can find this value in the API Console [Credentials page](https://console.developers.google.com/apis/credentials). <br />When developing **Cloudflare Workers**, there's no need to send this parameter. Just declare it in the `wrangler.toml` file as `GOOGLE_SECRET=`.
+	> Do not share your client secret to ensure the security of your app.
+- `scope`:
+	- Type: `string[]`.
+	- `Required`.
+	- Set of **permissions** to request the user's authorization to access your app for retrieving user information and performing actions on their behalf.<br /> Review all the scopes Google offers for utilizing their API on the [OAuth 2.0 Scopes page](https://developers.google.com/identity/protocols/oauth2/scopes). 
+	> If your app is not **verified** by Google, the accessible scopes for your app are significantly **limited**.
+- `response_type`:
+	- Type: `string`.
+	- `Required`.
+	- Choose the **OAuth flow** for the authentication process. The available options are:
+		- `code` for a more secure flow, applicable when the app's authentication flow is managed by the **server side**.
+		- `token` for quick access to the token, suitable when the app's authentication flow will be managed by the **client side**.
```

Comment:
Implicit grant type is not recommended, and is pretty much deprecated. I don't see a reason to implement this since Hono is a sever side library and it's intended for purely client side applications. If you were to implement this, `client_secret` should be optional

Reply:
Yeah It makes no sense, the OAuth flow should use the corresponding method to server side. Thanks!